### PR TITLE
Refactor address to nusnetId

### LINKED ISSUE #18: Refactor AddressBook: Change address field to nusnetId
- Rename field
- Add/modify regex validation

(change documentation as required)

### PR DESCRIPTION
**Pull request by theodoreleebrant**
TAskmaster does not need address, unlike address book. However, it needs
a unique identifier for a student, which is the nusnet ID.

The address field of the addressbook is therefore changed to nusnetID.
The validation of the nusnet ID is changed from the address validation.

Fixes #18

### CODE REVIEW

File: src/test/java/seedu/address/model/person/NusnetIDTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+public class NusnetIDTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new NusnetID(null));
+    }
+
+    @Test
+    public void constructor_invalidAddress_throwsIllegalArgumentException() {
```
Comment (jaya-r3) [inline src/test/java/seedu/address/model/person/NusnetIDTest.java]: Consider replacing address with NusnetId

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 2]
```diff
@@ -19,12 +19,12 @@
     public static final String DEFAULT_NAME = "Alice Pauline";
     public static final String DEFAULT_PHONE = "85355255";
     public static final String DEFAULT_EMAIL = "alice@gmail.com";
-    public static final String DEFAULT_ADDRESS = "123, Jurong West Ave 6, #08-111";
+    public static final String DEFAULT_ADDRESS = "e0000000";
```
Comment (jaya-r3) [inline src/test/java/seedu/address/testutil/PersonBuilder.java]: rename field

File: src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java [hunk 3]
```diff
@@ -64,10 +64,10 @@ public EditPersonDescriptorBuilder withEmail(String email) {
     }
 
     /**
-     * Sets the {@code Address} of the {@code EditPersonDescriptor} that we are building.
+     * Sets the {@code NusnetID} of the {@code EditPersonDescriptor} that we are building.
      */
     public EditPersonDescriptorBuilder withAddress(String address) {
```
Comment (jaya-r3) [inline src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java]: rename field

File: src/test/java/seedu/address/model/person/NusnetIDTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        assertThrows(NullPointerException.class, () -> NusnetID.isValidAddress(null));
+
+        // invalid addresses
+        assertFalse(NusnetID.isValidAddress("")); // empty string
+        assertFalse(NusnetID.isValidAddress(" ")); // spaces only
+
+        // valid addresses
+        assertTrue(NusnetID.isValidAddress("Blk 456, Den Road, #01-355"));
+        assertTrue(NusnetID.isValidAddress("-")); // one character
+        assertTrue(NusnetID.isValidAddress("Leng Inc; 1234 Market St; San Francisco CA 2349879; USA")); // long address
```
Comment (jaya-r3) [inline src/test/java/seedu/address/model/person/NusnetIDTest.java]: Please update testcases

File: src/test/java/seedu/address/model/person/NusnetIDTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class NusnetIDTest {
```
Comment (jaya-r3) [inline src/test/java/seedu/address/model/person/NusnetIDTest.java]: Please consider changing name to NusnetIdTest

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 6]
```diff
@@ -23,13 +23,13 @@
 public class ParserUtilTest {
     private static final String INVALID_NAME = "R@chel";
     private static final String INVALID_PHONE = "+651234";
-    private static final String INVALID_ADDRESS = " ";
+    private static final String INVALID_ADDRESS = "e1123456";
```
Comment (jaya-r3) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:26]: Consider renaming field

File: src/main/java/seedu/address/model/person/NusnetID.java [hunk 7]
```diff
@@ -5,35 +5,36 @@
 
 /**
  * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Guarantees: immutable; is valid as declared in {@link #isValidNusnetID(String)}
  */
-public class Address {
+public class NusnetID {
```
Comment (jaya-r3) [inline src/main/java/seedu/address/model/person/NusnetID.java]: Consider renaming class to NusnetId (coding standard)

**Review by jaya-r3 — CHANGES_REQUESTED** [review]
Looks good apart from:
- naming of 'NusnetID' class
- one test file needs to be updated
- some fields still have the 'address' name

**Comment by theodoreleebrant** [comment]
Updated according to coding standard and checkstyle. Updated testcases.

**Review by jaya-r3 — COMMENTED** [review]
LGTM, thanks for making the changes

**Review by jaya-r3 — APPROVED** [review]

**Pull request merged**